packages/components/managed-data-table/src/data-table-settings/density-manager/density-manager.spec.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { screen, render } from '../../../../../../test/test-utils';
+import DensityManager from '.';
+import PrimaryButton from '../../../../buttons/primary-button';
+import SecondaryButton from '../../../../buttons/secondary-button';
+import {
+  WRAPPED_TEXT_VISIBLE,
+  SHOW_HIDE_ON_DEMAND,
+  DENSITY_DEFAULT,
+  DENSITY_COMPACT,
+} from './constants';
+
+const createTestProps = (customProps) => ({
+  isCondensed: false,
+  isWrappingText: false,
+  onDensityDisplayChange: jest.fn(),
+  onTextWrappingChange: jest.fn(),
+  onClose: jest.fn(),
+  ...customProps,
+});
+
+describe('DensityManager', () => {
```

Comment:
I'm starting to wonder why do we have tests for each component, and not more "higher level" tests on the main component `ManagedDataTable`.

With RTL you usually want to test the UI feature altogether, and less the single UI elements of the feature.

Can you clarify maybe a bit this choice?